store/retry_layer.go

```diff
@@ -0,0 +1,4315 @@
+	BotStore
+	Root *RetryLayer
+}
+
+type RetryLayerChannelStore struct {
+	ChannelStore
+	Root *RetryLayer
+}
+
+type RetryLayerChannelMemberHistoryStore struct {
+	ChannelMemberHistoryStore
+	Root *RetryLayer
+}
+
+type RetryLayerClusterDiscoveryStore struct {
+	ClusterDiscoveryStore
+	Root *RetryLayer
+}
+
+type RetryLayerCommandStore struct {
+	CommandStore
+	Root *RetryLayer
+}
+
+type RetryLayerCommandWebhookStore struct {
+	CommandWebhookStore
+	Root *RetryLayer
+}
+
+type RetryLayerComplianceStore struct {
+	ComplianceStore
+	Root *RetryLayer
+}
+
+type RetryLayerEmojiStore struct {
+	EmojiStore
+	Root *RetryLayer
+}
+
+type RetryLayerFileInfoStore struct {
+	FileInfoStore
+	Root *RetryLayer
+}
+
+type RetryLayerGroupStore struct {
+	GroupStore
+	Root *RetryLayer
+}
+
+type RetryLayerJobStore struct {
+	JobStore
+	Root *RetryLayer
+}
+
+type RetryLayerLicenseStore struct {
+	LicenseStore
+	Root *RetryLayer
+}
+
+type RetryLayerLinkMetadataStore struct {
+	LinkMetadataStore
+	Root *RetryLayer
+}
+
+type RetryLayerOAuthStore struct {
+	OAuthStore
+	Root *RetryLayer
+}
+
+type RetryLayerPluginStore struct {
+	PluginStore
+	Root *RetryLayer
+}
+
+type RetryLayerPostStore struct {
+	PostStore
+	Root *RetryLayer
+}
+
+type RetryLayerPreferenceStore struct {
+	PreferenceStore
+	Root *RetryLayer
+}
+
+type RetryLayerReactionStore struct {
+	ReactionStore
+	Root *RetryLayer
+}
+
+type RetryLayerRoleStore struct {
+	RoleStore
+	Root *RetryLayer
+}
+
+type RetryLayerSchemeStore struct {
+	SchemeStore
+	Root *RetryLayer
+}
+
+type RetryLayerSessionStore struct {
+	SessionStore
+	Root *RetryLayer
+}
+
+type RetryLayerStatusStore struct {
+	StatusStore
+	Root *RetryLayer
+}
+
+type RetryLayerSystemStore struct {
+	SystemStore
+	Root *RetryLayer
+}
+
+type RetryLayerTeamStore struct {
+	TeamStore
+	Root *RetryLayer
+}
+
+type RetryLayerTermsOfServiceStore struct {
+	TermsOfServiceStore
+	Root *RetryLayer
+}
+
+type RetryLayerTokenStore struct {
+	TokenStore
+	Root *RetryLayer
+}
+
+type RetryLayerUserStore struct {
+	UserStore
+	Root *RetryLayer
+}
+
+type RetryLayerUserAccessTokenStore struct {
+	UserAccessTokenStore
+	Root *RetryLayer
+}
+
+type RetryLayerUserTermsOfServiceStore struct {
+	UserTermsOfServiceStore
+	Root *RetryLayer
+}
+
+type RetryLayerWebhookStore struct {
+	WebhookStore
+	Root *RetryLayer
+}
+
+func isRepeatableError(err error) bool {
+	var pqErr *pq.Error
+	var mysqlErr *mysql.MySQLError
+	switch {
+	case errors.As(errors.Cause(err), &pqErr):
+		if pqErr.Code == "40001" || pqErr.Code == "40P01" {
+			return true
+		}
+	case errors.As(errors.Cause(err), &mysqlErr):
+		if mysqlErr.Number == mySQLDeadlockCode {
+			return true
+		}
+	}
+	return false
+}
+
+func (s *RetryLayerAuditStore) Get(user_id string, offset int, limit int) (model.Audits, error) {
+
+	tries := 0
+	for {
+		resultVar0, resultVar1 := s.AuditStore.Get(user_id, offset, limit)
+		if resultVar1 == nil {
+			return resultVar0, resultVar1
+		}
```

Comment:
I understand this being generated code but would I wonder if it's worth making the naming a bit more human-friendly.
Can we assume the second returned value will always be an `error` and if there's only one it's also an `error` or does this logic break in some cases?

Reply:
Yes, that logic is not true, we can detect what is the error variable in the generator and name it accordingly, I can try to make the error variable always `err` instead of `resultVarN`. But we can't assume the position give any meaning, because is not true in all cases.